src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -725,7 +726,7 @@ private void UpdateGasBottleneckAndMarkForEviction(
 
                     if (!valid)
                     {
-                        MarkForEviction(tx, valid.AllowTxPoolReentrance);
+                        MarkForEviction(tx, valid.Error == TxErrorMessages.InvalidProofVersion);
```

Comment:
It breaks encapsulation, that additional flag did not affect comparison or any other previous logic

Reply:
Does it really? I don't see how.
Before: return this special type of error, add a flag, use the flag for special behavior
Now: return this special type of error, use special behavior on this special error